src/Neo/SmartContract/ApplicationEngine.cs

```diff
@@ -235,13 +248,13 @@ protected static void OnSysCall(ExecutionEngine engine, Instruction instruction)
         #endregion
 
         /// <summary>
-        /// Adds GAS to <see cref="GasConsumed"/> and checks if it has exceeded the maximum limit.
+        /// Adds GAS to <see cref="FeeConsumed"/> and checks if it has exceeded the maximum limit.
         /// </summary>
-        /// <param name="gas">The amount of GAS to be added.</param>
-        protected internal void AddGas(long gas)
+        /// <param name="datoshi">The amount of GAS, in the unit of datoshi, 1 datoshi = 1e-8 GAS, to be added.</param>
+        protected internal void AddFee(long datoshi)
         {
-            GasConsumed = checked(GasConsumed + gas);
-            if (GasConsumed > gas_amount)
+            FeeConsumed = GasConsumed = checked(FeeConsumed + datoshi);
```

Comment:
@Jim8y please remove the warning

Reply:
The warning is expected, marking and telling people that this is absolete. If we disable it here, people how use this will also not able to see it.

Reply:
But the warning will apif people use it, is not for us

Reply:
The warning shows to people exactly the way it shows us, during the build process, if we disable it here, it will also go disappear in users build terminal..... can be done easily, but i am not sure you would like that.